src/lib/viewers/controls/settings/SettingsDropdown.tsx

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import classNames from 'classnames';
+import uniqueId from 'lodash/uniqueId';
+import SettingsFlyout from './SettingsFlyout';
+import SettingsList from './SettingsList';
+import useClickOutside from '../hooks/useClickOutside';
+import { decodeKeydown } from '../../../util';
+import './SettingsDropdown.scss';
+
+export type ListItem = {
+    label: string;
+    value: string;
+};
+
+export type Props = {
+    className?: string;
+    label: string;
+    listItems: Array<ListItem>;
+    onSelect: () => void;
+    value?: string;
+};
+
+export default function SettingsDropdown({ className, label, listItems, onSelect, value }: Props): JSX.Element {
+    const { current: id } = React.useRef(uniqueId('bp-SettingsDropdown_'));
+    const buttonElRef = React.useRef<HTMLButtonElement | null>(null);
+    const dropdownElRef = React.useRef<HTMLDivElement | null>(null);
+    const listElRef = React.useRef<HTMLUListElement | null>(null);
+    const [isOpen, setIsOpen] = React.useState(false);
+
+    const handleButtonClick = (): void => setIsOpen(!isOpen);
+    const handleClickOutside = (): void => setIsOpen(false);
+    const handleKeyDown = (event: React.KeyboardEvent): void => {
+        const key = decodeKeydown(event);
+
+        if (key === 'Escape') {
+            setIsOpen(false);
+
+            if (buttonElRef.current) {
+                buttonElRef.current.focus(); // Prevent focus from falling back to the body on flyout close
+            }
+        }
+
+        if (key === 'ArrowUp' || key === 'ArrowDown' || key === 'Escape') {
+            event.preventDefault();
+            event.stopPropagation();
+        }
+    };
+    const handleSelect = (selectedOption: string): void => {
+        setIsOpen(false);
+        onSelect(selectedOption);
+    };
+    const createClickHandler = (selectedOption: string) => (event: React.KeyboardEvent): void => {
```

Comment:
Should this be a MouseEvent? I'm confused why TS isn't erroring on both this issue and the DOMRect height/width properties.